python-sdk/src/astro/databases/google/bigquery.py

```diff
@@ -212,6 +215,56 @@ def merge_table(
             statement += f" WHEN MATCHED THEN {update_statement}"
         self.run_sql(sql_statement=statement)
 
+    def is_native_autodetect_schema_available(  # skipcq: PYL-R0201
+        self, file: File  # skipcq: PYL-W0613
+    ) -> bool:
+        """
+        Check if native auto detection of schema is available.
+
+        :param file: File used to check the file type of to decide
+            whether there is a native auto detection available for it.
+        """
+        is_file_type_supported = (
+            file.type.name in NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_TYPES
+        )
+        is_file_location_supported = (
+            file.location.location_type
+            in NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_LOCATIONS
+        )
+        return is_file_type_supported and is_file_location_supported
+
+    def create_table_using_native_schema_autodetection(
+        self,
+        table: Table,
+        file: File,
+    ) -> None:
+        """
+        Create a SQL table, automatically inferring the schema using the given file via native database support.
```

Comment:
@feluelle I think this would populate the file in the table as well right?

If the user has given a pattern that results in two files

example - 
pattern : s3://tmp/

which results in files:
1. s3://tmp/test1.csv
2. s3://tmp/test2.csv

With the existing code of autodetecting schema, we will end up with:

Step 1: Autoschema detect -  Table + 1st file load
Step 2: Load data in a table - 1st file load + 2nd file load

Would it result in 1st file being loaded twice? Can we confirm if this is not the case?

cc: @sunank200

Reply:
> Step 1: Autoschema detect - Table + 1st file load

The autoschema detect should not load the file actually..

Reply:
But good point, I will check that.

Reply:
I tested it for snowflake by adding:
```python
    statement = f"SELECT COUNT(*) FROM {database.get_table_qualified_name(table)}"
    count = database.run_sql(statement).scalar()
    assert count == 0
```
which passed.

Reply:
@feluelle Nice, Can we add one for bigquery as well?

Reply:
I have added it see https://github.com/astronomer/astro-sdk/pull/780#discussion_r968293149

Reply:
@kaxil, @utkarsharma2  I think @feluelle is truncating the table: https://github.com/astronomer/astro-sdk/blob/main/python-sdk/src/astro/databases/google/bigquery.py#L279

The test only needs fix